go-controller/pkg/util/pod_annotation.go

```diff
@@ -116,6 +117,19 @@ func MarshalPodAnnotation(annotations map[string]string, podInfo *PodAnnotation,
 	for _, ip := range podInfo.IPs {
 		pa.IPs = append(pa.IPs, ip.String())
 	}
+
+	existingPa, ok := podNetworks[nadName]
+	if ok {
+		if len(pa.IPs) != len(existingPa.IPs) {
+			return nil, ErrOverridePodIPs
+		}
+		for _, ip := range pa.IPs {
```

Comment:
there are some users who totally copy pod yaml definition including the annotation bits accidentally over; so I think with this change we are establishing that if there was an existing pod-networks annotation with an IP allocated in it that doesn't match the new one we are getting, we will error out always.. right?

Reply:
yes, and I think we need to error it out. otherwise, the cni side might use the incorrect Pod IP to bringup Pod, right?